src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/GpNameAndAddress.java

```diff
@@ -0,0 +1,59 @@
+package uk.nhs.digital.nhsconnect.nhais.model.edifact;
+
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import uk.nhs.digital.nhsconnect.nhais.exceptions.EdifactValidationException;
+
+import org.springframework.util.StringUtils;
+
+/**
+ * Example NAD+GP+2750922,295:900'
+ */
+@Getter @Setter @RequiredArgsConstructor
```

Comment:
I think "Required" works with `private final` and here you don't have any. Or does the `@NonNull` also makes it a constructor parameter?

Reply:
From javadoc of `@RequiredArgsConstructor`:

>  Required arguments are final fields and fields with constraints such as @NonNull.